src/Slider/range.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2015 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import { useEffect } from 'react';
+
+import { isFactor } from './factor';
+
+export type Values = readonly number[];
+export type Range = {
+    min: number;
+    max: number;
+    decimals?: number;
+    step?: number; // positive number
+};
+
+export type RangeOrValues = Range | Values;
+
+export const isValues = (
+    rangeOrValues: RangeOrValues
+): rangeOrValues is Values => Array.isArray(rangeOrValues);
+
+export const getMin = (rangeOrValues: RangeOrValues) =>
+    isValues(rangeOrValues) ? rangeOrValues[0] : rangeOrValues.min;
+
+export const getMax = (rangeOrValues: RangeOrValues) =>
+    isValues(rangeOrValues)
+        ? rangeOrValues[rangeOrValues.length - 1]
+        : rangeOrValues.max;
+
+export const getStep = (range: Range) =>
+    range.step != null ? range.step : 0.1 ** (range.decimals ?? 0);
```

Comment:
Consider to avoid floating point number issues: 
```suggestion
    range.step != null ? range.step : Number(0.1 ** range.decimals ?? 0).toFixed(range.decimals ?? 0);
```

As 

`Number(0.1 ** 3).toFixed(3) = 0.001`

while

`0.1 ** 3 = 0.0010000000000000002`